### DISCUSSION TITLE
rke2 automated upgrades and longhorn

### DISCUSSION THREAD

**Discussion by linus365** [longhorn/longhorn#6285] [Q&A] [answered]
Hey,

I have longhorn running on a 3-node rke2 cluster.
I don't want to perform rke2 upgrades manually all the time and I want to consider using the automated upgrades feature provided by rke2: https://docs.rke2.io/upgrade/automated_upgrade

But I am hesitant so far to just leave all nodes enabled for on auto upgrades and usually manually patch the node labels one by one allowing the ugprade operator to bump rke.
This way I can make sure that the volumes are fully healthy on longhorn before triggering the next node to upgrade.
I really don't want a situation where I loose all three replicas of my PVs.

Has anyone tried something similar before?

Excited too see any input here :) Have a great evening

**Comment by yangchiu** [comment, marked as answer]
Although not using `rke2`, we tested k8s automated upgrade using `k3s (system upgrade controller)` and `rancher` before in https://github.com/longhorn/longhorn/issues/3304#issuecomment-1467174481

You can find related topics at:

_2. Upgrade Kubernetes for k3s cluster with standalone System Upgrade Controller deployment_
_3. Upgrade Kubernetes for imported k3s cluster in Rancher_
_4. Upgrade Kubernetes for provisioned k3s cluster in Rancher_

But it needs to be noticed that the `concurrency` should still be set to **1** to upgrade node one by one instead of trying to upgrade all the nodes at once.

Hope it helps!

**Reply by innobead** [reply]
https://longhorn.io/docs/1.5.0/volumes-and-nodes/maintenance/#in-place-upgrade

**Reply by linus365** [reply]
Thanks for all the info! I might be brave enough now to attempt the auto-update
